src/cbmc/README.md

```diff
@@ -11,3 +11,106 @@ others) to create a goto-program, `goto-symex` to unwind the loops the
 given number of times and to produce and equation system and finally
 `solvers` to find a counter-example (technically, `goto-symex` is then
 used to construct the counter-example trace).
+
+## Architecture
+
+CBMC operates in three phases:
+
+1. Parse command-line options; convert passed source files into a GOTO model
+2. Run `goto-symex` to turn the GOTO model into a SAT or SMT equation
+3. Solve the equation and report results
+
+The GOTO model is generated using `initialize_goto_model`; see documentaton
+in the `goto-programs/` directory. Goto-symex is similarly documented in its
+own directory.
+
+## Equation solving
```

Comment:
Shouldn't references to the documentation @johanneskloos is working on also be provided here? It's good that the documentation here focuses on the reports generated and the options, but the reader may be curious about the details of doing bounded model checking.

Reply:
Unsure how to do that exactly, but sure. @johanneskloos any hints?

Reply:
The general approach behind most of the steps is documented in background concepts (background-concepts should work for a link); in particular, translation to an IR (not quite GOTO, but in the same general spirit) is in IR_section, while steps 2 and 3 are described in BMC_section (referencing the SSA section). Does this help?